From martinez.javier@gmail.com  Thu Aug 12 13:07:55 2010
Subject: Staging: spectra: removes q->prepare_flush_fn, fix build breakage
From: Javier Martinez Canillas <martinez.javier@gmail.com>
To: Greg KH <greg@kroah.com>, Randy Dunlap <randy.dunlap@oracle.com>,  devel@driverdev.osuosl.org, David Woodhouse <David.Woodhouse@intel.com>,  kernel-janitors@vger.kernel.org
Date: Fri, 09 Jul 2010 23:25:12 -0400
Message-ID: <1278732312.1725.17.camel@lenovo>

This patch is the first one of a patchset that allows 
stagin/spectra driver to compile in linux-next. 

blk_queue_ordered doesn't receive a prepare flush function anymore

Signed-off-by: Javier Martinez Canillas <martinez.javier@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/spectra/ffsport.c |   10 +---------
 1 file changed, 1 insertion(+), 9 deletions(-)

--- a/drivers/staging/spectra/ffsport.c
+++ b/drivers/staging/spectra/ffsport.c
@@ -272,13 +272,6 @@ static int get_res_blk_num_os(void)
 	return res_blks;
 }
 
-static void SBD_prepare_flush(struct request_queue *q, struct request *rq)
-{
-	rq->cmd_type = REQ_TYPE_LINUX_BLOCK;
-	/* rq->timeout = 5 * HZ; */
-	rq->cmd[0] = REQ_LB_OP_FLUSH;
-}
-
 /* Transfer a full request. */
 static int do_transfer(struct spectra_nand_dev *tr, struct request *req)
 {
@@ -650,8 +643,7 @@ static int SBD_setup_device(struct spect
 	/* Here we force report 512 byte hardware sector size to Kernel */
 	blk_queue_logical_block_size(dev->queue, 512);
 
-	blk_queue_ordered(dev->queue, QUEUE_ORDERED_DRAIN_FLUSH,
-					SBD_prepare_flush);
+	blk_queue_ordered(dev->queue, QUEUE_ORDERED_DRAIN_FLUSH);
 
 	dev->thread = kthread_run(spectra_trans_thread, dev, "nand_thd");
 	if (IS_ERR(dev->thread)) {
